### PR TITLE
[Port] Doing mail again

### PR DESCRIPTION
**Pull request by adamsong**
Steals the mail code from #12948 because cool pr that went dormant

# Document the changes in your pull request

Ports https://github.com/tgstation/tgstation/pull/58146

- Mail boxes show up in cargo shuttle
- Mail has credits for the people addressed on the letter
- Mail can have job-dependent gifts too
- Cargo can deliver mail if they want or tell people to fetch it
- Some holidays are mail holidays, meaning no mail on them
- Can't open someone elses mail

[Weights and chances for loots](https://gist.github.com/adamsong/69be6a80fcb3eab09e09612853a1c9fc)

# Spriting
Sprites for mail and stuff

# Wiki Documentation

Mail arrives on shuttle, needs removed before the shuttle can leave

# Changelog

:cl:    adamsong, Inari-Whitebear, ArcaneMusic, Jaw-sh, Tralezab
rscadd: Mail for cargo
wip: This is work in progress
imageadd: mail images and mail crate image
/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
YES MAIL I LOVE IT

File: code/modules/jobs/job_types/cargo_technician.dm [hunk 1]
```diff
@@ -21,6 +21,14 @@
 	display_order = JOB_DISPLAY_ORDER_CARGO_TECHNICIAN
 	minimal_character_age = 18 //We love manual labor and exploiting the young for our corporate purposes
 
+	mail_goodies = list(
+		/obj/item/pizzabox = 10,
+		/obj/item/stack/sheet/mineral/gold = 5,
+		/obj/item/stack/sheet/mineral/uranium = 4,
+		/obj/item/stack/sheet/mineral/diamond = 3,
+		/obj/item/gun/ballistic/rifle/boltaction = 1
```
Comment (TheRyeGuyWhoWillNowDie) [inline code/modules/jobs/job_types/cargo_technician.dm:29]: fitting a 6-foot long bolt action rifle into an envelope 🤔

**Comment by adamsong** [comment]
Added a verb to print a list of all goodies, their weights and relative percents, as well as highlight any jobs with no job specific loot. I'll use this at some point to finalize loot tables

**Review by cuackles — APPROVED** [review]
try to @ the original sprite artist if posible

**Comment by adamsong** [comment]
All the authors from the original pr are credited, unsure of who did sprites specifically

**Comment by CMOisLing** [comment]
There should be some way of stealing mail if your a traitor cargo tech.

**Comment by adamsong** [comment]
Thats illegal

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**